src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/Security/SecureProcessConfiguration.cs

```diff
@@ -0,0 +1,70 @@
+using System.Runtime.InteropServices;
+using System.Collections.Generic;
+using System;
+
+namespace Azure.Tools.GeneratorAgent.Security
+{
+    /// <summary>
+    /// Provides secure, hardcoded process execution configuration for cross-platform support.
+    /// No user-configurable values to prevent security vulnerabilities.
+    /// </summary>
+    internal static class SecureProcessConfiguration
+    {
+        /// <summary>
+        /// Gets the PowerShell Core executable name (cross-platform).
+        /// </summary>
+        public static string PowerShellExecutable => "pwsh";
+        
+        /// <summary>
+        /// Gets the Node.js executable name.
+        /// </summary>
+        public static string NodeExecutable => "node";
+        
+        /// <summary>
+        /// Gets the NPM executable name.
+        /// On Windows, use PowerShell to execute npm for reliable execution. On other platforms, use npm.
+        /// </summary>
+        public static string NpmExecutable => RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? PowerShellExecutable : "npm";
```

Comment:
This form will run every time, even though platform won't change.

```csharp
public static string NpmExecutable { get; } =  RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? PowerShellExecutable : "npm";
```

That form will execute once to initialize and then use the cached value.   Since you can't change platforms while running, I'd suggest converting them over.